`wasi` incorrectly defined on global

### Version

v18.12.1

### Platform

Linux - 5.4.190-107.353.amzn2.x86_6

### Subsystem

wasi global

### What steps will reproduce the bug?

`global.wasi`

```typescript
Object.getOwnPropertyNames(global).forEach((k) => {
    const v = Reflect.get(global, k);
    if (isConstructor(v)) {
        NATIVE_CLASS_PROTOTYPES.set(v.prototype, v);
    }
});
```

`Uncaught Error: Cannot find module 'wasi'`

### How often does it reproduce? Is there a required condition?

Always

### What is the expected behavior?

`global.wasi` evaluates to undefined;

```javascript
assert(global.hasOwnProperty('wasi') === false)
```

### What do you see instead?

`Uncaught Error: Cannot find module 'wasi'` for `global.wasi`

```javascript
assert(global.hasOwnProperty('wasi') === true)
```

### Additional information

`--experimental-wasi-unstable-preview1` not specified

--- a/lib/internal/main/repl.js
+++ b/lib/internal/main/repl.js
@@ -8,7 +8,6 @@ const {
   markBootstrapComplete
 } = require('internal/process/pre_execution');
 
-const esmLoader = require('internal/process/esm_loader');
 const {
   evalScript
 } = require('internal/process/execution');
@@ -37,6 +36,7 @@ if (process.env.NODE_REPL_EXTERNAL_MODULE) {
     process.exit(kGenericUserError);
   }
 
+  const esmLoader = require('internal/process/esm_loader');
   esmLoader.loadESM(() => {
     console.log(`Welcome to Node.js ${process.version}.\n` +
       'Type ".help" for more information.');

--- a/test/parallel/test-repl-built-in-modules.js
+++ b/test/parallel/test-repl-built-in-modules.js
@@ -30,3 +30,19 @@ assert.doesNotMatch(
   stdout,
   /Uncaught Error: Cannot find module 'wasi'[\w\W]+- <repl>\n/);
 assert.match(stdout, /{ WASI: \[class WASI\] }/);
+
+{
+  const res = cp.execFileSync(process.execPath, ['-i'], {
+    input: "'wasi' in global",
+    encoding: 'utf8',
+  });
+  // `wasi` shouldn't be defined on global when the flag is not set
+  assert.match(res, /false\n/);
+}
+{
+  const res = cp.execFileSync(process.execPath, ['-i', '--experimental-wasi-unstable-preview1'], {
+    input: "'wasi' in global",
+    encoding: 'utf8',
+  });
+  assert.match(res, /true\n/);
+}

